cocos/render-scene/core/pass.ts

```diff
@@ -258,6 +258,10 @@ export class Pass {
         const type = Pass.getTypeFromHandle(handle);
         const ofs = Pass.getOffsetFromHandle(handle);
         const block = this._getBlockView(type, binding);
+        if (DEBUG) {
+            const validator = type2validator[type];
+            assertID(validator && validator(value), 12011, binding, Type[type]);
```

Comment:
After adding validation check, will it break the compatibility of user code? 
If so, we need to add documentation for this change.

Reply:
Yes, that's also the reason we must report clear error message for type mismatch.
We can add a note to the release note